targettest/sym1/symDisplay.c

```diff
@@ -0,0 +1,356 @@
+               DISPLAY.d5 = DISP_V_1;
+               DISPLAY.d6 = DISP_V_2;
+               break;
+            case 'W':
+               DISPLAY.d0 = DISPLAY.d1;
+               DISPLAY.d1 = DISPLAY.d2;
+               DISPLAY.d2 = DISPLAY.d3;
+               DISPLAY.d3 = DISPLAY.d4;
+               DISPLAY.d4 = DISPLAY.d5;
+               DISPLAY.d5 = DISP_W_1;
+               DISPLAY.d6 = DISP_W_2;
+               break;
+            case 'w':
+               DISPLAY.d0 = DISPLAY.d1;
+               DISPLAY.d1 = DISPLAY.d2;
+               DISPLAY.d2 = DISPLAY.d3;
+               DISPLAY.d3 = DISPLAY.d4;
+               DISPLAY.d4 = DISPLAY.d5;
+               DISPLAY.d5 = DISP_W_1;
+               DISPLAY.d6 = DISP_W_2;
+               break;
+            case 'Y':
+               DISPLAY.d6 = DISP_Y;
+               break;
+            case 'y':
+               DISPLAY.d6 = DISP_Y;
+               break;
+            case 'Z':
+               DISPLAY.d6 = DISP_Z;
+               break;
+            case 'z':
+               DISPLAY.d6 = DISP_Z;
+               break;
+            case ' ':
+               DISPLAY.d6 = DISP_SPACE;
+               break;
+            case '.':
+               DISPLAY.d6 = DISP_PERIOD;
+               break;
+            case '-':
+               DISPLAY.d6 = DISP_HYPHEN;
+               break;
+            case '\'':
+               DISPLAY.d6 = DISP_APOSTR;
+               break;
+            case '"':
+               DISPLAY.d6 = DISP_APOSTR;
+               break;
+            case '=':
+               DISPLAY.d6 = DISP_EQUAL;
+               break;
+            case '_':
+               DISPLAY.d6 = DISP_BOTTOM;
+               break;
+            case '/':
+               DISPLAY.d6 = DISP_SLASH;
+               break;
+            case '\\':
+               DISPLAY.d6 = DISP_BACKSLASH;
+               break;
+            default:
+               displayable = 0;        // Character not mapped
+         }
+
+         if( displayable ) {
+
+            putchar( buffer[l] );      // Send it to the console
+
+            DISPLAY.d0 = DISPLAY.d1;   // Scroll to the left
+            DISPLAY.d1 = DISPLAY.d2;
+            DISPLAY.d2 = DISPLAY.d3;
+            DISPLAY.d3 = DISPLAY.d4;
+            DISPLAY.d4 = DISPLAY.d5;
+            DISPLAY.d5 = DISPLAY.d6;
+
+            for( d = 0; d < flashes ; d++ ) {
+               fdisp();                // Display
+            }
+         }
+      }
+
+      for( e = 0; e < 6; e++ ) {       // Gradually fill the
+         DISPLAY.d0 = DISPLAY.d1;      // display with spaces
+         DISPLAY.d1 = DISPLAY.d2;
+         DISPLAY.d2 = DISPLAY.d3;
+         DISPLAY.d3 = DISPLAY.d4;
+         DISPLAY.d4 = DISPLAY.d5;
+         DISPLAY.d5 = DISP_SPACE;
+         DISPLAY.d6 = DISP_SPACE;
+         for( d = 0; d < flashes ; d++ ) {
+            fdisp();                   // Display
+         }
+      }
+   }
+
+   puts( "\n\nEnjoy your day!\n\n" );
+}
```

Comment:
Please adhere to the cc65 code style in general, as given by the two excamples above.

Reply:
Updated.  Please check and confirm.

Reply:
Sorry, for being not precise enough.

I wasn't talking about main() returning a value. It's perfectly fine that main() doesn't return a value for programs specific for a target where there's no ue for the return value of main().

I'm (still) talking about source code style. In cc65 it isn't `puts( "\n\nEnjoy your day!\n\n" );`. It's `puts ("\n\nEnjoy your day!\n\n");` .

Reply:
Oh, I see now.  OK, I'll make changes to my source files that reflect that sort of whitespace convention.

As an aside, I used to write code like that.  Actually, I did for many years.  But on the latest project I've been worlking on, the style is like what you saw.  So it has become a habit for the past few years.

Anyway, I'll go modify all the targettest samples to use the style you have described.

I'm going to leave leave the 'int main' and 'return int' convention.  It is sort of six one way and a half-dozen the other, since there is no operating system to capture the return code.  But it doesn't hurt anything either.  The Sym-1 monitor program will have a nice clean zero in the accumulator.  :-)